### PR TITLE
Improving Electron tests

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

Improves running the Electron tests:
- Producing a cjs bundle of the app importer.
- Upgrading Electron to v24.
- Loading realm in the renderer process still fails.

### CODE REVIEW

File: packages/bindgen/src/templates/node-wrapper.ts [hunk 1]
```diff
@@ -52,10 +52,10 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
     };
   `);
   nodeLines.push(`
-    /*global global*/
+    /*global global, require */
     import { createRequire } from 'node:module';
-    const require = createRequire(import.meta.url);
-    const nativeModule = require("./realm.node");
+    const nodeRequire = typeof require === 'function' ? require : createRequire(import.meta.url);
```
Comment (elle-j) [inline packages/bindgen/src/templates/node-wrapper.ts:57]: Unsure if this is being linted.
```suggestion
    const nodeRequire = typeof require === "function" ? require : createRequire(import.meta.url);
```

**Review by elle-j — APPROVED** [review]
🙂

File: integration-tests/environments/electron/app/main.js [hunk 2]
```diff
@@ -19,6 +19,9 @@
 // This file is pretty much a copy of https://github.com/electron/electron-quick-start/blob/master/main.js
 const electron = require("electron");
 const { app, BrowserWindow } = electron;
+// Needed to forward console logs
+const remote = require("@electron/remote/main");
+remote.initialize();
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply integration-tests/environments/electron/app/main.js:24]: Only when executed from the `@electron/remote`, which this is not.

File: packages/bindgen/src/templates/node-wrapper.ts (hunk 1 above)
Comment (kraenhansen) [reply packages/bindgen/src/templates/node-wrapper.ts:57]: Unfortunately, it's a limitation of the TS in TS code, that it isn't being linted 😞

**Pull request merged**